src/api/routes/bns/names.ts

```diff
@@ -39,5 +42,39 @@ export function createBNSNamesRouter(db: DataStore): RouterWithAsync {
     }
   });
 
+  router.getAsync('/', async (req, res) => {
+    const page = parsePagingQueryInput(req.query.page ?? 0);
+
+    const { results } = await db.getNamesList({ page });
+    if (results.length === 0 && req.query.page) {
+      res.status(400).json(BNSErrors.InvalidPageNumber);
+    }
+    res.json(results);
+  });
+
+  router.getAsync('/:name', async (req, res) => {
+    const { name } = req.params;
+
+    const nameQuery = await db.getName({ name });
+    if (!nameQuery.found) {
+      return res.status(404).json({ error: `cannot find name ${name}` });
+    }
+
+    const { result } = nameQuery;
+    const nameInfo: BNSGetNameInfoResponse = {
+      address: result.address,
+      blockchain: bnsBlockchain,
+      expire_block: result.expire_block,
+      grace_period: result.grace_period,
+      last_txid: result.tx_id ? result.tx_id : '',
+      resolver: result.resolver,
+      status: result.status ? result.status : '',
+      zonefile: result.zonefile,
+      zonefile_hash: result.zonefile_hash,
+    };
+    const response = Object.fromEntries(Object.entries(nameInfo).filter(([_, v]) => v != null));
```

Comment:
`// TODO: loose null comparison hard to reason about`

Reply:
Here I am removing the keys which have null values in the `BNSGetNameInfoResponse` object. 
If you can suggest a better approach that would be great or should I add a comment for the logic. 
@zone117x

Reply:
There's a loose comparison `!=` against null. I would expect this should be disabled by the linter, because it's not always obvious what js values can be coerced into.